components/storage/blockdevice/COMPONENT_SPIF/SPIFBlockDevice.h

```diff
@@ -38,6 +38,13 @@ enum spif_bd_error {
 #define SPIF_MAX_REGIONS    10
 #define MAX_NUM_OF_ERASE_TYPES 4
 
+/*
+ * To enable add '"spif-block.debug": true' to your mbed_app.json
+ */
+#ifndef MBED_CONF_SPIF_BLOCK_DEBUG
+#define MBED_CONF_SPIF_BLOCK_DEBUG false
+#endif
```

Comment:
Why is this needed? Is the `debug_if()` macro sensitive of macros being not defined?

Reply:
Yes, `debug_if` is a [function](https://github.com/ARMmbed/mbed-os/blob/e69a7c913d9148b1bbec80e18d4b4c5048ba426e/platform/mbed_debug.h#L63). Code won't compile if MBED_CONF_SPIF_BLOCK_DEBUG is not defined.

Reply:
I think this should not be neccessary?

Just change the mbed_lib.json to only accept values 1 and 0.
For. ex. `"options": [0, 1].`

Just to clarify for general information:
Setting value `0 `in mbed_lib.json defines the MACRO with value of `0`.
Setting value `null `in mbed_lib.json does not define any MACRO.

Reply:
Thanks, @teetak01 , you are right. I amended the code.